src/bonsaimodule.c

```diff
@@ -12,6 +12,32 @@ PyObject *LDAPEntryObj = NULL;
 PyObject *LDAPValueListObj = NULL;
 char debugmod = 0;
 
+#if !defined(WIN32) && !defined(MACOSX) && LDAP_VENDOR_VERSION > 20443
+/* The asynchronous connection build does not function properly on
+   macOS and only works on other unix systems from version 2.4.44 */
+int async_conn = 1;
```

Comment:
Tiny memory saving if this variable is a char. I'd change the macro condition to set this variable 1 on every system and 0 on Mac by default.